Adding AAD authentication client & sample, updating README.md file (#40596)
Azure/azure-sdk-for-net: sdk/translation/Azure.AI.Translation.Text/README.md

TextTranslationClient client = new TextTranslationClient(new AzureKeyCredential(apiKey), new Uri(endpoint), region);
```

#### Create `TextTranslationClient` with Entra Credential

> Client API key authentication is used in most of the examples, but you can also authenticate with Entra (Azure Active Directory) using the [Azure Identity library][azure_identity].  To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below, please install the Azure.Identity package:

Reviewer: ```suggestion
Client API key authentication is used in most of the examples, but you can also authenticate with Entra ID (Azure Active Directory) using the [Azure Identity library][azure_identity].  To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below, please install the Azure.Identity package:
```

> Client API key authentication is used in most of the examples, but you can also authenticate with Entra ID (Azure Active Directory) using the [Azure Identity library][azure_identity].  To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below, please install the Azure.Identity package:

Reviewer: ```suggestion
Client API key authentication is used in most of the examples, but you can also authenticate with Microsoft Entra ID using the [Azure Identity library][azure_identity]. To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below, install the Azure.Identity package:
```

```dotnetcli
dotnet add package Azure.Identity
```

> Create a [custom subdomain][custom_subdomain] for your resource in order to use this type of authentication.  Use this value for the `endpoint` variable for `Text Translator Custom Endpoint`.  Only custom subdomains are supported by the SDK.  Outside of the SDK you can use AAD authorization for Machine Tranlsation endpoints as detailed [here][mt_endpoints].

Reviewer: ```suggestion
Create a [custom subdomain][custom_subdomain] for your resource in order to use this type of authentication.  Use this value for the `endpoint` variable for `Text Translator Custom Endpoint`.  Only custom subdomains are supported by the SDK.  Outside of the SDK, you can use Entra authorization for Machine Translation endpoints, as detailed [here][mt_endpoints].
```

> You will also need to [register a new AAD application][register_aad_app] and [grant access][aad_grant_access] to your Translator resource by assigning the `"Cognitive Services User"` role to your service principal.  Additional information about Entra authentication is available [here][custom_details].

Reviewer: ```suggestion
You will also need to [register a new Entra application][register_aad_app] and [grant access][aad_grant_access] to your Translator resource by assigning the `"Cognitive Services User"` role to your service principal.  Additional information about Entra authentication is available [here][custom_details].
```

> Set the values of the `client ID`, `tenant ID`, and `client secret` of the AAD application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.  The DefaultAzureCredential constructor uses these variables to create your credentials.

Reviewer: ```suggestion
Set the values of the `client ID`, `tenant ID`, and `client secret` of the Entra application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.  The DefaultAzureCredential constructor uses these variables to create your credentials.
```

[dictionaryexamples_sample]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/translation/Azure.AI.Translation.Text/samples/Sample6_DictionaryExamples.md

[translator_resource_create]: https://learn.microsoft.com/azure/cognitive-services/Translator/create-translator-resource
[azure_identity]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/README.md
[DefaultAzureCredential]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md#defaultazurecredential

> [register_aad_app]: https://docs.microsoft.com/azure/cognitive-services/authentication#assign-a-role-to-a-service-principal
> [aad_grant_access]: https://docs.microsoft.com/azure/cognitive-services/authentication#assign-a-role-to-a-service-principal
> [custom_subdomain]: https://docs.microsoft.com/azure/cognitive-services/authentication#create-a-resource-with-a-custom-subdomain

Reviewer: ```suggestion
[register_aad_app]: https://learn.microsoft.com/azure/cognitive-services/authentication#assign-a-role-to-a-service-principal
[aad_grant_access]: https://learn.microsoft.com/azure/cognitive-services/authentication#assign-a-role-to-a-service-principal
[custom_subdomain]: https://learn.microsoft.com/azure/cognitive-services/authentication#create-a-resource-with-a-custom-subdomain
```
